### PR TITLE
Download save-demo-agent to pod

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
* Moved curl building command logic to `AgentDownloadUtils.kt`
* Implemented startup command logic that uses `AgentDownloadUtils` in order to build startup command
* Added `/download-agent` endpoint to `DependencyController` of `save-demo`
* Updated configuration on `demo-configmap.yaml`

### CODE REVIEW

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/AgentDownloadUtils.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * Utils for building curl command that downloads save-agent and save-demo-agent to pod
+ */
+
+package com.saveourtool.save.utils
+
+import org.intellij.lang.annotations.Language
+
+@Language("bash")
+private val defaultShellOptions: Array<out String> = arrayOf(
```
Comment (kgevorkyan) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/AgentDownloadUtils.kt:11]: may be sequence? anyway you do such cast in `getShellOptions`

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt [hunk 2]
```diff
@@ -122,4 +125,16 @@ class DependencyController(
         "Could not find demo for $organizationName/$projectName."
     }
         .flatMapMany { dependencyStorage.archive(it.organizationName, it.projectName, version) }
+
+    /**
+     * @return save-demo-agent.kexe as [Resource] wrapped into [Mono]
+     */
+    @GetMapping("/download-agent", produces = [MediaType.APPLICATION_OCTET_STREAM_VALUE])
+    fun downloadSaveAgent(): Mono<out Resource> = run {
+        val executable = "save-demo-agent.kexe"
```
Comment (kgevorkyan) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:134]: may be use this variable as argument? in that way we can probably reuse this logic for `save-agent` too
if `save-agent` have something similar

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt [hunk 3]
```diff
@@ -122,4 +125,16 @@ class DependencyController(
...
+
+        downloadFromClasspath(executable) {
+            "Can't find $executable"
```
Comment (kgevorkyan) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:137]:
```suggestion
            "Can't find $executable on classpath"
```

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt (hunk 2 above)
Comment (kgevorkyan) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:134]: or at least can be reused as `SAVE_DEMO_AGENT_EXECUTABLE_NAME`

Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:134]: Probably the most of this logic will be refactored in #1867 so I guess it should be done after it is merged

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/AgentDownloadUtils.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/AgentDownloadUtils.kt:11]: Yep, will do it

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt (hunk 2 above)
Comment (kgevorkyan) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:134]: Ok, add some `todo` in code, then

Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:134]: Will do it

Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:136]: Done

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/AgentDownloadUtils.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/AgentDownloadUtils.kt]: Changed `Array<out String>` to `Sequence<String>`

**Review by kgevorkyan — APPROVED** [review]

**Pull request merged**